src/hole.rs

```diff
@@ -264,6 +257,17 @@ fn deallocate(mut hole: &mut Hole, addr: usize, mut size: usize) {
                 hole = move_helper(hole).next_unwrap(); // start next iteration at next hole
                 continue;
             }
+            _ if hole_addr + hole.size == addr => {
```

Comment:
Is there a reason why you moved this match case to the bottom? I think it should work too, but I'm not 100% sure… 

(The order matters for some cases, e.g. the `XXXFFFFYYYYY` case must come before the `XXXFFFF__YYYYY` and `XXX__FFFFYYYYY` cases.)